docs/guides/js-waku/use-waku-react.md

```diff
@@ -0,0 +1,356 @@
+} from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+import { utf8ToBytes } from "@waku/sdk";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node, error, isLoading } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create an encoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { encoder  } = useCreateContentPair(contentTopic);
+
+	// Wait for the node to finish loading before sending messages
+	// (isLoading === false)
+
+	// Bind push method to a node and encoder
+	const { push } = useLightPush({ node, encoder });
+
+	// Send the message using Light Push
+	const sendMessage = async (text) => {
+		if (!push || !text) {
+			return;
+		}
+		const payload = utf8ToBytes(text);
+		await push({ payload });
+	};
+	sendMessage("Hello, World!");
+}
+```
+
+:::info
+Wait for the node to finish loading before sending messages (`isLoading` === `false`).
+:::
+
+## Receive Messages Using Filter
+
+Use the `useFilterMessages()` hook to receive messages from a `Filter` subscription:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useFilterMessages,
+} from "@waku/react";
+import { Protocols } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Receive messages from Filter subscription
+	const { error, messages, isLoading } = useFilterMessages({ node, decoder });
+	console.log(messages);
+
+	// "error" captures any error that occurs while receiving messages
+	// "messages" contains a list of messages the subscription received
+	// "isLoading" indicates whether the node is still subscribing to Filter
+	// Use these to handle states and render the UI in your application
+}
+```
+
+## Retrieve Messages Using Store
+
+Use the `useStoreMessages()` hook to retrieve messages from the `Store` protocol:
+
+```js title="App.js"
+import {
+	useCreateLightNode,
+	useCreateContentPair,
+	useStoreMessages,
+} from "@waku/react";
+import { Protocols, PageDirection } from "@waku/interfaces";
+
+function App() {
+	// Create and start a Light Node and wait for remote peers
+	const { node } = useCreateLightNode({
+		options: { defaultBootstrap: true },
+		protocols: [Protocols.LightPush, Protocols.Filter],
+	});
+
+	// Choose a content topic and create a decoder
+	const contentTopic = "/waku-react-guide/1/message/utf8";
+	const { decoder } = useCreateContentPair(contentTopic);
+
+	// Set the query options
+	const options = {
+		pageDirection: PageDirection.BACKWARD,
+		pageSize: 20,
```

Comment:
I believe `pageSize` is optimized for 10 on nwaku side.
@jm-clius can you please confirm? What about for PostgreSQL @Ivansete-status ?

Don't show the usage of this specific option. let the dev use the default options.

Reply:
Afaik recommended page size is indeed 20. We do limit the number of content topics per query to 10, mostly to reduce complexity in big databases such as those for Status. This is less relevant for postgresql than sqlite.